improver/lapse_rate.py

```diff
@@ -326,12 +326,43 @@ def _create_heightdiff_mask(self, all_orog_subsections):
 
         height_diff = np.subtract(self.all_orog_subsections, central_points)
         height_diff = np.absolute(height_diff)
-
-        height_diff_mask = np.where(height_diff >= self.max_height_diff, True,
+        height_diff_mask = np.where(height_diff >=
+                                    self.max_height_diff, True,
                                     False)
 
         return height_diff_mask
 
+    def _generate_heightdiff_mask(self, all_orog_subsections):
+        """
+        Function to create a mask for any neighbouring points where the height
+        difference from the central point is greater than max_height_diff.
+
+        Slice through the orography subsection array to remove central
+        points.
+        Extracts the height value of each central point and masks out the
+        neighbouring points where their height difference is greater than
+        the maximum.
+
+        Args:
+            all_orog_subsections(2D numpy.ndarray):
+               Each row contains the height values of each neighbourhood.
+
+        Returns:
+            generator:
```

Comment:
My understanding is that this function is a generator and it returns an iterable object. See https://mypy.readthedocs.io/en/stable/kinds_of_types.html#generators for how to express this in mypy type syntax. Suggest reviewing documented function return type.

Reply:
Okay, I've updated it.